src/max_encoded_len.rs

```diff
@@ -32,6 +32,10 @@ use crate::alloc::sync::Arc;
 pub trait MaxEncodedLen: Encode {
 	/// Upper bound, in bytes, of the maximum encoded size of this item.
 	fn max_encoded_len() -> usize;
+	/// Upper bound, in bytes, of the maximum encoded size of the compact encoding of this item.
+	fn max_compact_encoded_len() -> usize {
+		Self::max_encoded_len()
+	}
```

Comment:
Hmm... personally I'd probably prefer to keep this as a `#[no_doc]` and make it an internal implementation detail, instead of making it public.

Reply:
by internal you mean simply adding the `#[doc(hidden)]` attribute or is there other things you can do to truly keep it internal?

Reply:
> by internal you mean simply adding the `#[doc(hidden)]` attribute or is there other things you can do to truly keep it internal?

Just hide it. I guess you could also prefix its name, e.g. prepend `scale_internal_` to the name (which I did when I was adding `scale_internal_decode_bytes` on `Input`)